# This is my first attempt at analyzing datas, I wasn't experienced yet and that's why I mainly used lists and wrote long chapters of code

Here I analyzed the datas of violent deaths in US from 2012 to 2014, trying to verify if there is a connection with the race of the people involved and with the type of death.

In [2]:
# Reading the dataset

import csv
f=open("guns.csv","r")
csv_reader=csv.reader(f)
data=list(csv_reader)


In [3]:
# Showing the first 5 rows

headers=data[0]
data_new=data[1:]
print(headers)
print(data_new[:5])

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [21]:
# Counting the amounts of death per year

year_counts={}
for i in data_new:
    year=i[1]
    year_counts[year]=0
for i in data_new:
    year=i[1]
    year_counts[year]+=1

print(year_counts)

{'2012': 33563, '2014': 33599, '2013': 33636}


In [22]:
# Extracting the dates of every single event

date=[]
for num in data_new:
    import datetime
    anno=int(num[1])
    mese=int(num[2])
    dates=datetime.datetime(year=anno, month=mese, day=1)
    date.append(dates)  
   

In [23]:
# Counting how many deaths per month have happened

date_counts={}

jan12=0
feb12=0
mar12=0
apr12=0
may12=0
jun12=0
jul12=0
aug12=0
sep12=0
oct12=0
nov12=0
dec12=0
jan13=0
feb13=0
mar13=0
apr13=0
may13=0
jun13=0
jul13=0
aug13=0
sep13=0
oct13=0
nov13=0
dec13=0
jan14=0
feb14=0
mar14=0
apr14=0
may14=0
jun14=0
jul14=0
aug14=0
sep14=0
oct14=0
nov14=0
dec14=0
for elem in date:
    if elem.year==2012 and elem.month==1:
        jan12+=1
        date_counts["Jan12"]=jan12
    if elem.year==2012 and elem.month==2:
        feb12+=1
        date_counts["Feb12"]=feb12
    if elem.year==2012 and elem.month==3:
        mar12+=1
        date_counts["Mar12"]=mar12
    if elem.year==2012 and elem.month==4:
        apr12+=1
        date_counts["Apr12"]=apr12
    if elem.year==2012 and elem.month==5:
        may12+=1
        date_counts["May12"]=may12
    if elem.year==2012 and elem.month==6:
        jun12+=1
        date_counts["Jun12"]=jun12
    if elem.year==2012 and elem.month==7:
        jul12+=1
        date_counts["Jul12"]=jul12
    if elem.year==2012 and elem.month==8:
        aug12+=1
        date_counts["Aug12"]=aug12
    if elem.year==2012 and elem.month==9:
        sep12+=1
        date_counts["Sep12"]=sep12
    if elem.year==2012 and elem.month==10:
        oct12+=1
        date_counts["Oct12"]=oct12
    if elem.year==2012 and elem.month==11:
        nov12+=1
        date_counts["Nov12"]=nov12
    if elem.year==2012 and elem.month==12:
        dec12+=1
        date_counts["Dec12"]=dec12
        
    if elem.year==2013 and elem.month==1:
        jan13+=1
        date_counts["Jan13"]=jan13
    if elem.year==2013 and elem.month==2:
        feb13+=1
        date_counts["Feb13"]=feb13
    if elem.year==2013 and elem.month==3:
        mar13+=1
        date_counts["Mar13"]=mar13
    if elem.year==2013 and elem.month==4:
        apr13+=1
        date_counts["Apr13"]=apr13
    if elem.year==2013 and elem.month==5:
        may13+=1
        date_counts["May13"]=may13
    if elem.year==2013 and elem.month==6:
        jun13+=1
        date_counts["Jun13"]=jun13
    if elem.year==2013 and elem.month==7:
        jul13+=1
        date_counts["Jul13"]=jul13
    if elem.year==2013 and elem.month==8:
        aug13+=1
        date_counts["Aug13"]=aug13
    if elem.year==2013 and elem.month==9:
        sep13+=1
        date_counts["Sep13"]=sep13
    if elem.year==2013 and elem.month==10:
        oct13+=1
        date_counts["Oct13"]=oct13
    if elem.year==2013 and elem.month==11:
        nov13+=1
        date_counts["Nov13"]=nov13
    if elem.year==2013 and elem.month==12:
        dec13+=1
        date_counts["Dec13"]=dec13  
        
    if elem.year==2014 and elem.month==1:
        jan14+=1
        date_counts["Jan14"]=jan14
    if elem.year==2014 and elem.month==2:
        feb14+=1
        date_counts["Feb14"]=feb14
    if elem.year==2014 and elem.month==3:
        mar14+=1
        date_counts["Mar14"]=mar14
    if elem.year==2014 and elem.month==4:
        apr14+=1
        date_counts["Apr14"]=apr14
    if elem.year==2014 and elem.month==5:
        may14+=1
        date_counts["May14"]=may14
    if elem.year==2014 and elem.month==6:
        jun14+=1
        date_counts["Jun14"]=jun14
    if elem.year==2014 and elem.month==7:
        jul14+=1
        date_counts["Jul14"]=jul14
    if elem.year==2014 and elem.month==8:
        aug14+=1
        date_counts["Aug14"]=aug14
    if elem.year==2014 and elem.month==9:
        sep14+=1
        date_counts["Sep14"]=sep14
    if elem.year==2014 and elem.month==10:
        oct14+=1
        date_counts["Oct14"]=oct14
    if elem.year==2014 and elem.month==11:
        nov14+=1
        date_counts["Nov14"]=nov14
    if elem.year==2014 and elem.month==12:
        dec14+=1
        date_counts["Dec14"]=dec14
        
print(date_counts) 
sum_months=jan12+feb12+mar12+apr12+may12+jun12+jul12+aug12+sep12+oct12+nov12+dec12+jan13+feb13+mar13+apr13+may13+jun13+jul13+aug13+sep13+oct13+nov13+dec13+jan14+feb14+mar14+apr14+may14+jun14+jul14+aug14+sep14+oct14+nov14+dec14
print("total by month:",sum_months)

{'Jan12': 2758, 'Oct12': 2733, 'Nov12': 2729, 'Aug12': 2954, 'Jan14': 2651, 'Apr13': 2798, 'May13': 2806, 'May14': 2864, 'Jan13': 2864, 'Feb12': 2357, 'Oct14': 2865, 'Dec14': 2857, 'Sep13': 2742, 'Mar13': 2862, 'Aug14': 2970, 'Nov14': 2756, 'Jun12': 2826, 'Jul13': 3079, 'Dec12': 2791, 'Jul14': 2884, 'Sep12': 2852, 'Jun14': 2931, 'Sep14': 2914, 'Aug13': 2859, 'Nov13': 2758, 'Feb14': 2361, 'May12': 2999, 'Jul12': 3026, 'Apr12': 2795, 'Mar14': 2684, 'Jun13': 2920, 'Mar12': 2743, 'Oct13': 2808, 'Dec13': 2765, 'Feb13': 2375, 'Apr14': 2862}
total by month: 100798


In [24]:
# Counting how many deaths by gender and race have happened

sex_counts={}
race_counts={}
M=0
F=0
Black=0
White=0
Asian_Pacific_Islander=0
Native_American_Native_Alaskan=0
Hispanic=0
races=[]
for c in data_new:
    if c[5]=="M":
        M+=1
        sex_counts["M"]=M
    if c[5]=="F":
        F+=1
        sex_counts["F"]=F
    if c[7]=="Black":
        Black+=1
        race_counts["Black"]=Black
    if c[7]=="White":
        White+=1
        race_counts["White"]=White
    if c[7]=="Asian/Pacific Islander":
        Asian_Pacific_Islander+=1
        race_counts["Asian_Pacific_Islander"]=Asian_Pacific_Islander
    if c[7]=="Native American/Native Alaskan":
        Native_American_Native_Alaskan+=1
        race_counts["Native_American_Native_Alaskan"]=Native_American_Native_Alaskan
    if c[7]=="Hispanic":
        Hispanic+=1
        race_counts["Hispanic"]=Hispanic
        
print(sex_counts)
print(race_counts)
print("total by gender:", M+F)
print("total by race:", Black+White+Hispanic+Asian_Pacific_Islander+Native_American_Native_Alaskan)

        
        

{'F': 14449, 'M': 86349}
{'Hispanic': 9022, 'White': 66237, 'Native_American_Native_Alaskan': 917, 'Black': 23296, 'Asian_Pacific_Islander': 1326}
total by gender: 100798
total by race: 100798


The total amount of deaths between the years 2012-2014 is 100798.
Analysing the numbers month-wise,there seems to be a common trend showing that the peak of crime rates was reached during the month of July (exceeding the 3000 units) both in 2012 and in 2013.
Gender-wise the vast majority of crimes has been committed by men (86349(m) against 14449(f)).
Race-wise the "leading" ethnicity is the Caucasian/White with a total amount of 66237 committed crimes. Less than half of those crimes were committed by Black people (23296) who are followed, in this special chart, by the Hispanics having committed less than 10000 crimes (9022). Lastly we find the Asian/Pacific Islanders (1326) and the Native American/Native Alaskan with just 917 crimes in a three years span of time.


Now we will compare the results we found with the Census of the population in US.

In [25]:
g=open("census.csv","r")
csv_reader_census=csv.reader(g)
census=list(csv_reader_census)
census=census[1]
print(census)

['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']


In [26]:
# Counting the total population of the single races in US

mapping={}
mapping=race_counts
mapping["Asian_Pacific_Islander"]=int(census[15])+int(census[14])
mapping["Native_American_Native_Alaskan"]=int(census[13])
mapping["White"]=int(census[10])
mapping["Black"]=int(census[12])
mapping["Hispanic"]=int(census[11])
print(mapping)
race_per_hundredk={}

{'Hispanic': 44618105, 'White': 197318956, 'Native_American_Native_Alaskan': 3739506, 'Black': 40250635, 'Asian_Pacific_Islander': 15834141}


In [27]:
# Calculating the death rate per 100k people

race_per_hundredk["White"]=(White/int(census[10]))*100000
race_per_hundredk["Black"]=(Black/int(census[12]))*100000
race_per_hundredk["Hispanic"]=(Hispanic/int(census[11]))*100000
race_per_hundredk["Asian_Pacific_Islander"]=(Asian_Pacific_Islander/(int(census[15])+int(census[14])))*100000
race_per_hundredk["Native_American_Native_Alaskan"]=(Native_American_Native_Alaskan/int(census[13]))*100000
print(race_per_hundredk)

{'Hispanic': 20.220491210910907, 'White': 33.56849303419181, 'Native_American_Native_Alaskan': 24.521955573811088, 'Black': 57.8773477735196, 'Asian_Pacific_Islander': 8.374309664161762}


So the rates for deaths by guns in US per a hundred thousand people is 57.87 for black people, 24.52 for native americans, 33.56 for white people, 8.37 for asian and pacific islanders and 20.22 for hispanics

In [35]:
# Repeating the same analysis just for homicides

homicide_race_counts={}
homicide_race_counts["White"]=0
homicide_race_counts["Black"]=0
homicide_race_counts["Hispanic"]=0
homicide_race_counts["Native_American_Native_Alaskan"]=0
homicide_race_counts["Asian_Pacific_Islander"]=0
for h in data_new:
    if h[7]=="White" and h[3]=="Homicide":
        homicide_race_counts["White"]+=1
for h in data_new:
    if h[7]=="Black" and h[3]=="Homicide":
        homicide_race_counts["Black"]+=1
for h in data_new:
    if h[7]=="Hispanic" and h[3]=="Homicide":
        homicide_race_counts["Hispanic"]+=1
for h in data_new:
    if h[7]=="Asian/Pacific Islander" and h[3]=="Homicide":
        homicide_race_counts["Asian_Pacific_Islander"]+=1
for h in data_new:
    if h[7]=="Native American/Native Alaskan" and h[3]=="Homicide":
        homicide_race_counts["Native_American_Native_Alaskan"]+=1
        
print(homicide_race_counts)
print("total homicides:", (homicide_race_counts["White"]+homicide_race_counts["Black"]+homicide_race_counts["Hispanic"]+homicide_race_counts["Native_American_Native_Alaskan"]+homicide_race_counts["Asian_Pacific_Islander"]))

{'Hispanic': 5634, 'White': 9147, 'Asian_Pacific_Islander': 559, 'Black': 19510, 'Native_American_Native_Alaskan': 326}
total homicides: 35176


So considering just the murders happened between 2012 and 2014, the race who committed most of them is the black race with a total amount of 19510, followed by the white race with 9147 (less than half of the murders made by black people). Hispanic's homicide rates is even lower (5634) and finally the Asian/Pacific Islander community and the Native American/Alaskan one have a similar total amount, respectively 559 and 326.

In [36]:
# Calculating the homicide rate per month 

h_jan=0
h_feb=0
h_mar=0
h_apr=0
h_may=0
h_jun=0
h_jul=0
h_aug=0
h_sep=0
h_oct=0
h_nov=0
h_dec=0
homicide_per_month={}
for b in data_new:
    if b[2]=="01" and b[3]=="Homicide":
        h_jan+=1
        homicide_per_month["Jan"]=h_jan
    if b[2]=="02" and b[3]=="Homicide":
        h_feb+=1
        homicide_per_month["Feb"]=h_feb
    if b[2]=="03" and b[3]=="Homicide":
        h_mar+=1
        homicide_per_month["Mar"]=h_mar
    if b[2]=="04" and b[3]=="Homicide":
        h_apr+=1
        homicide_per_month["Apr"]=h_apr
    if b[2]=="05" and b[3]=="Homicide":
        h_may+=1
        homicide_per_month["May"]=h_may
    if b[2]=="06" and b[3]=="Homicide":
        h_jun+=1
        homicide_per_month["Jun"]=h_jun
    if b[2]=="07" and b[3]=="Homicide":
        h_jul+=1
        homicide_per_month["Jul"]=h_jul
    if b[2]=="08" and b[3]=="Homicide":
        h_aug+=1
        homicide_per_month["Aug"]=h_aug
    if b[2]=="09" and b[3]=="Homicide":
        h_sep+=1
        homicide_per_month["Sep"]=h_sep
    if b[2]=="10" and b[3]=="Homicide":
        h_oct+=1
        homicide_per_month["Oct"]=h_oct
    if b[2]=="11" and b[3]=="Homicide":
        h_nov+=1
        homicide_per_month["Nov"]=h_nov
    if b[2]=="12" and b[3]=="Homicide":
        h_dec+=1
        homicide_per_month["Dec"]=h_dec
print(homicide_per_month)
print("total homicides:", (h_jan+h_feb+h_mar+h_apr+h_may+h_jun+h_jul+h_aug+h_sep+h_oct+h_nov+h_dec))

{'Nov': 2919, 'Feb': 2178, 'Apr': 2845, 'Oct': 2968, 'Jun': 3130, 'Jul': 3269, 'Mar': 2780, 'May': 2976, 'Sep': 2966, 'Dec': 3191, 'Jan': 2829, 'Aug': 3125}
total homicides: 35176


As observed before the homicide rates throughout a year appears to be obscillating but it shows a pretty clear peak in July (3269), it is followed by December (3191), June (3130) and August (3125). Which means that basically homicides happen during the summer or christmas holidays. When people are together conflicts tend to rise and homicides occurr.

In [40]:
# Calculating the homicide rate per gender

h_m=0
h_f=0
homicide_per_gender={}
for d in data_new:
    if d[5]=="M" and d[3]=="Homicide":
        h_m+=1
        homicide_per_gender["M"]=h_m
    if d[5]=="F" and d[3]=="Homicide":
        h_f+=1
        homicide_per_gender["F"]=h_f
print(homicide_per_gender)
print("total homicides:", (h_m+h_f))

{'F': 5373, 'M': 29803}
total homicides: 35176


So the vast majority of the murderers are men with a total amount being almost six times the number of homicides committed by women: 29803 vs 5373.

In [43]:
# Repeating the same analysis just for accidents

acc_gender={}
acc_race={}
acc_m=0
acc_f=0
acc_wh=0
acc_bl=0
acc_as=0
acc_his=0
acc_nat=0
for r in data_new:
    if r[3]=="Accidental" and r[5]=="M":
        acc_m+=1
        acc_gender["M"]=acc_m
    if r[3]=="Accidental" and r[5]=="F":
        acc_f+=1
        acc_gender["F"]=acc_f
    if r[3]=="Accidental" and r[7]=="White":
        acc_wh+=1
        acc_race["White"]=acc_wh
    if r[3]=="Accidental" and r[7]=="Black":
        acc_bl+=1
        acc_race["Black"]=acc_bl
    if r[3]=="Accidental" and r[7]=="Hispanic":
        acc_his+=1
        acc_race["Hispanic"]=acc_his
    if r[3]=="Accidental" and r[7]=="Asian/Pacific Islander":
        acc_as+=1
        acc_race["Asian/Pacific Islander"]=acc_as
    if r[3]=="Accidental" and r[7]=="Native American/Native Alaskan":
        acc_nat+=1
        acc_race["Native American/Native Alaskan"]=acc_nat
print(acc_gender, acc_m+acc_f)
print(acc_race, acc_wh+acc_bl+acc_as+acc_his+acc_nat)

{'F': 218, 'M': 1421} 1639
{'Hispanic': 145, 'White': 1132, 'Asian/Pacific Islander': 12, 'Black': 328, 'Native American/Native Alaskan': 22} 1639


As it happens for homicides, even when talking about accidental deaths the number of male victims is almost 7 times bigger than the female victims ones (1421 vs 218).

Moreover, race-wise, we can state that people killed by accidental events are mostly whites (1132). Their number is 4 times bigger than the blacks' one (328). Then we find the hispanics (145). The remaining two communities (Asian/Pacific Islanders and Native American/Alaskan) have very low numbers, respectively 12 and 22.

In [46]:
# Repeating the analysis with respect to the place of the murder and the education of the murderer

place=[]
education=[]
for z in data_new:
    place.append(z[9])
    education.append(z[10])
    
print(set(place))
print(set(education))

{'Home', 'School/instiution', 'Other unspecified', 'Trade/service area', 'Other specified', 'Sports', 'Industrial/construction', 'Street', 'Residential institution', 'NA', 'Farm'}
{'1', '4', '5', '3', '2', 'NA'}


In [4]:
deaths_loc={}
deaths_edu={}
loc_ho=0
loc_sc=0
loc_ot=0
loc_tr=0
loc_otsp=0
loc_sp=0
loc_ind=0
loc_st=0
loc_res=0
loc_na=0
loc_fa=0
edu_1=0
edu_2=0
edu_3=0
edu_4=0
edu_5=0
edu_na=0
for t in data_new:
    if t[9]=='Home':
        loc_ho+=1
        deaths_loc['Home']=loc_ho
    if t[9]=='School/instiution':
        loc_sc+=1
        deaths_loc['School/instiution']=loc_sc
    if t[9]=='Other unspecified':
        loc_ot+=1
        deaths_loc['Other unspecified']=loc_ot
    if t[9]=='Trade/service area':
        loc_tr+=1
        deaths_loc['Trade/service area']=loc_tr
    if t[9]=='Other specified':
        loc_otsp+=1
        deaths_loc['Other specified']=loc_otsp
    if t[9]=='Sports':
        loc_sp+=1
        deaths_loc['Sports']=loc_sp
    if t[9]=='Industrial/construction':
        loc_ind+=1
        deaths_loc['Industrial/construction']=loc_ind
    if t[9]=='Street':
        loc_st+=1
        deaths_loc['Street']=loc_st
    if t[9]=='Residential institution':
        loc_res+=1
        deaths_loc['Residential institution']=loc_res
    if t[9]=='NA':
        loc_na+=1
        deaths_loc['NA']=loc_na
    if t[9]=='Farm':
        loc_fa+=1
        deaths_loc['Farm']=loc_fa
    if t[10]=="1":
        edu_1+=1
        deaths_edu["1"]=edu_1
    if t[10]=="2":
        edu_2+=1
        deaths_edu["2"]=edu_3
    if t[10]=="3":
        edu_3+=1
        deaths_edu["3"]=edu_3
    if t[10]=="4":
        edu_4+=1
        deaths_edu["4"]=edu_4
    if t[10]=="5":
        edu_5+=1
        deaths_edu["5"]=edu_5
    if t[10]=="NA":
        edu_na+=1
        deaths_edu["NA"]=edu_na
        
print(deaths_loc)
print(deaths_edu)

{'Other unspecified': 8867, 'Home': 60486, 'Industrial/construction': 248, 'Street': 11151, 'Trade/service area': 3439, 'NA': 1384, 'Sports': 128, 'Residential institution': 203, 'Other specified': 13751, 'Farm': 470, 'School/instiution': 671}
{'3': 21680, '1': 21823, 'NA': 53, '2': 21680, '4': 12946, '5': 1369}


So basically the vast majority of the gun deaths happens at home. Approximately 60% of the total(60486 on 100798).
Moreover it seems that those deaths are equally distributed among the people with a Less than high school, high school or some college education. People graduated from college tend not to be victims 